tests/basic/pure_test.go

```diff
@@ -1862,3 +1943,231 @@ var _ = Describe("{CreateAndDeleteMultipleVolumesInParallel}", func() {
+					Parameters:        params,
+					ReclaimPolicy:     &reclaimPolicyDelete,
+					VolumeBindingMode: &bindMode,
+				}
+
+				k8sStorage := storage.Instance()
+				_, err = k8sStorage.CreateStorageClass(&scObj)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("verify sc [%s] creation", scName))
+			}
+
+			createNs := func(nsName string) {
+				ns := &v1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: nsName,
+					},
+				}
+				log.InfoD("Creating namespace %v", nsName)
+				_, err = core.Instance().CreateNamespace(ns)
+
+				if err != nil {
+					if apierrors.IsAlreadyExists(err) {
+						log.Infof("Namespace %s already exists. Skipping creation.", ns.Name)
+					} else {
+						log.FailOnError(err, fmt.Sprintf("error creating namespace [%s]", nsName))
+					}
+				}
+			}
+
+			createPVC := func(pvcName, scName, appNs string, errCh chan error, wg *sync.WaitGroup) {
+				defer wg.Done()
+				log.InfoD("creating PVC [%s] in namespace [%s]", pvcName, appNs)
+
+				pvcObj := &v1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      pvcName,
+						Namespace: appNs,
+					},
+					Spec: v1.PersistentVolumeClaimSpec{
+						AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+						StorageClassName: &scName,
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceStorage: resource.MustParse("5Gi"),
+							},
+						},
+					},
+				}
+				_, err = core.Instance().CreatePersistentVolumeClaim(pvcObj)
+				if err != nil {
+					errCh <- err
+				}
+
+			}
+
+			createSC(scName)
+			createNs(nsName)
+			stNodes := node.GetStorageDriverNodes()
+			var wg sync.WaitGroup
+			errCh := make(chan error, numPVCs+len(stNodes)) // creating a buffered channel with length for worst case scenario failures
+			for i := 1; i <= numPVCs; i++ {
+				pvcName := fmt.Sprintf("%s-%d", pvcPrefix, i)
+				wg.Add(1)
+				go createPVC(pvcName, scName, nsName, errCh, &wg)
+			}
+
+			//restarting all volume driver nodes sequentially
+			wg.Add(1)
+			go func(errCh chan error, wg *sync.WaitGroup) {
+				defer wg.Done()
+
+				for _, stNode := range stNodes {
+					restartNodes := []node.Node{stNode}
+					err = Inst().V.StopDriver(restartNodes, false, nil)
+					if err != nil {
+						errCh <- err
+						break
+					}
+					err = Inst().V.WaitDriverDownOnNode(stNode)
+					if err != nil {
+						errCh <- err
+						break
+					}
+					err = Inst().V.StartDriver(stNode)
+					if err != nil {
+						errCh <- err
+						break
+					}
+					err = Inst().V.WaitDriverUpOnNode(stNode, 5*time.Minute)
+					if err != nil {
+						errCh <- err
+						break
+					}
+					break //remove it while raising PR
+				}
+			}(errCh, &wg)
+
+			wg.Wait()
+			close(errCh)
+
+			if len(errCh) > 0 {
+				for err := range errCh {
+					log.Errorf("%v", err)
+				}
+				log.FailOnError(fmt.Errorf("error(s) occured while creating PVC and restartign PX on nodes"), "no errors should occuer")
```

Comment:
*restarting PX on nodes
no errors should *occur

Reply:
fixed